common/main/java/com/couchbase/lite/AbstractDatabase.java

```diff
@@ -1497,12 +1497,13 @@ private void saveResolvedDocumentWithFlags(
                 }
             }
             else {
-                final FLSliceResult mergedBody = resolvedDoc.encode();
-                // if the resolved doc has attachments, be sure has the flag
-                if (C4Document.dictContainsBlobs(mergedBody, sharedKeys)) {
-                    mergedFlags |= C4Constants.RevisionFlags.HAS_ATTACHMENTS;
+                try (FLSliceResult mergedBody = resolvedDoc.encode()) {
+                    // if the resolved doc has attachments, be sure it has the flag
+                    if (C4Document.dictContainsBlobs(mergedBody, sharedKeys)) {
+                        mergedFlags |= C4Constants.RevisionFlags.HAS_ATTACHMENTS;
+                    }
+                    mergedBodyBytes = mergedBody.getContent();
```

Comment:
Expect that getContent() will copy the bytes?

Reply:
It does.